src/rewards/poprewards_cache.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2019-2020 Xenios SEZC
+// https://www.veriblock.org
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#include <veriblock/rewards/poprewards.hpp>
+#include <veriblock/rewards/poprewards_cache.hpp>
+
+namespace altintegration {
+
+PopRewardsBigDecimal PopRewardsCache::calculateDifficulty(
+    const BlockIndex<AltBlock>& /* ignore */) const {
+  auto cached = buffer.asVector();
+  // remove last element - it is the current block that we calculate difficulty for
+  cached.pop_back();
+  PopRewardsBigDecimal sumscore = 0.0;
+  size_t count = 0;
+  for (const auto& c : reverse_iterate(cached)) {
```

Comment:
if you need `asVector` just to iterate correctly, maybe you can just expose `rbegin` and `rend` in `buffer`, and then use them to reverise iterate?

also, why reverse iteration? I think total sum will be equal anyway